src/App.tsx

```diff
@@ -1,61 +1,116 @@
+          username: 'unknown',
+          email: 'unknown@example.com',
+        },
+    } 
+
+    setTodos(previous => [...previous, newTodo]);
+    setTitle('');
+    setUserId(0);
+    return;
+  }
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={onSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+        <label htmlFor="titleInput">Todo title</label>
+          <input type="text" 
+          data-cy="titleInput"  
+          value={title}
+          onChange={(event)=> {
+            setTitle(event.target.value)
+            if (titleError) setTitleError(false)
+          }}
+          placeholder='enter title'
+          />
+          {titleError && <span className="error">Please enter a title</span>}
+          
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
+          <label htmlFor="userSelect">User</label>
+          <select data-cy="userSelect" value={userId} onChange={ 
+            (event)=>{
+              setUserId(+event.target.value)
+              if (userError) {
+                setUserError(false)
+              }
+            }
+            
+            }>
+          <option value="0">
               Choose a user
             </option>
+            {usersFromServer.map(user => {
+              
+                return(<option value= {user.id} key = {user.id}>
+                  {user.name}
+                </option>)
+              
+            })}
+            
           </select>
-
-          <span className="error">Please choose a user</span>
+            {userError && <span className="error">Please choose a user</span>}
+          
         </div>
 
         <button type="submit" data-cy="submitButton">
           Add
         </button>
       </form>
 
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <TodoList todos={todos}/>
```

Comment:
Ensure that each `TodoInfo` component within the `TodoList` has a unique `key` prop. This is crucial for React's reconciliation process and helps prevent potential rendering issues.